HIP/hip-415.md

```diff
@@ -0,0 +1,255 @@
+It would work as follows:
+- Skip if not `testnet` or `mainnet`
+- Get the last entry in the mapping file and look it up in the `record_file` table.
+- If the row's block number matches file then there's nothing to do.
+- If the row's block number does not match, update it and update every row before it and after relative to that number.
+
+After the migration, Mirror nodes must be able to answer queries of type:
+- Getting block by hash/number
+- Getting a list of transactions for a block by number/hash
+- Getting logs based on block number filtering
+
+### Block Properties
+
+The following table specifies all `block` properties and at which point they will be computed. Mirror nodes must `hex` encode all properties prior to exposing them through their APIs. This table defines the properties that must be returned through APIs from Mirror Nodes.
+
+| Property | Computed By | Description  |
+| --- | --- | --- |
+| number | Services | Stored in services state (only the current number) and exported in the record stream. It is the consecutive number of the record file that is being incremented by 1 for every new record file. The number will be initially set in services through the bootstrapping process. It is exposed to the 1) EVM during Transaction Execution through the NUMBER opcode; 2) as a new property in the record file and ingested by mirror nodes. |
+| timestamp | Services | Stored in services state (only the last block timestamp) and computed by services. It is the consensusTimestamp of the first transaction in the record file. It is exposed to the 1) EVM during Transaction Execution through the TIMESTAMP opcode; 2) implicitly exported in the record file through the TS of the first transaction in the record file. |
+| hash | Services | Stored in services (last 256 blocks). It is the 32 byte prefix of the runningHash of the previous record file. That is the running hash of the last record stream object from the previous record file. It is exposed to the 1) EVM during Transaction Execution through the BLOCKHASH opcode; 2) In the record file as the End Object Running Hash |
+| baseFeePerGas | Mirror Node | Always zero, since there is no EIP-1559 style floating block capacity fees in Hedera. |
+  | difficulty | Mirror Node | Hardcoded by Mirror Node(s) to hex encoded 0 |
+  | extraData | Mirror Node | Hardcoded by Mirror Nodes(s) to 0x |
+  | gasLimit | Mirror Node |  Computed by Mirror Node(s). The gas throttle limit per second multiplied by the target block time. |
+  | gasUsed | Mirror Node | Computed by Mirror Node(s). The sum of the gasUsed value for all ContractCall and ContractCreate transactions within the block. |
+  | logsBloom | Mirror Node | Computed by Mirror Node(s). It is the bloom filter for the logs within the block. |
+  | miner | Mirror Node | Hardcoded by the Mirror Nodes to the 0.0.98 address in the long zero prefix format.  |
```

Comment:
Not sure we should populate this as in the future staking accounts 800/801 will also receive block rewards.

Reply:
Miner implies two things, (a) who gets paid and (b) who decided what transactions went into that block.  Since hedera doesn't really have a single party picking transactions and ordering them for any timeframe an argument can be made there is no miner, so an all-zero address may be more correct and dodges the 0.0.98, 0.0.600, 0.0.601 question.

Reply:
Your argument is valid. I was focused on the (a) part and haven't thought about the (b) part. I am okay with setting it to `0.0.0`